cmp/grouping/GroupingChooserModel.ts

```diff
@@ -343,4 +336,41 @@ export class GroupingChooserModel extends HoistModel {
         if (this.persistFavorites) ret.favorites = this.favorites;
         return ret;
     }
+
+    //------------------------
+    // Implementation
+    //------------------------
+    private normalizeDimensions(
+        dims: Array<DimensionSpec | string>
+    ): Record<string, DimensionSpec> {
+        dims = dims ?? [];
+        const ret = {};
+        dims.forEach(it => {
+            const dim = this.createDimension(it);
+            ret[dim.name] = dim;
+        });
+        return ret;
+    }
+
+    private createDimension(src: DimensionSpec | string) {
+        src = isString(src) ? {name: src} : src;
+        throwIf(
+            !src.hasOwnProperty('name'),
+            "Dimensions provided as Objects must define a 'name' property."
+        );
+        return {displayName: genDisplayName(src.name), ...src};
+    }
+
+    private cleanStaleDims() {
+        const {value, dimensionNames, allowEmpty} = this,
+            cleanValue = value.filter(dim => dimensionNames.includes(dim));
+
+        if (isEqual(value, cleanValue)) return;
+
+        if (isEmpty(cleanValue) && !allowEmpty) {
+            cleanValue.push(dimensionNames[0]);
+        }
```

Comment:
We made the choice here to reset the value to "the first dimension in the list" in the event that a dim change would leave it an unsupported state.  

We don't think that this will happen often, but felt that we needed to do *something* in this case.

It was difficult to know how to make handling of possible value changes and possible dimension changes consistent.  E.g. if an invalid value is set - where value contains dim not in dimensions - we just log a warning (invisible to user) and reject the value change.  

Rejecting the dim change (because value held something not in the new list) did not seem appropriate in the same way.  We could throw an exception instead, but in the end felt better to prioritize accepting and proceeding with the control's overall "setup" at the expense of any given/transient value it might be holding.